### PR TITLE
[Locked] Fixed #306 - weird exception on demo website

### PR DESCRIPTION
**Pull request by mickaelandrieu**
If a Response has been already sent in defaultAction(), this should not be returned again in invokeAction()

### CODE REVIEW

**Comment by eric-chau** [comment]
Nice catch =) :+1:

**Comment by eric-chau** [comment]
Don't merge it now we need to validate some behavior first.

**Comment by mickaelandrieu** [comment]
Yes, I suggest a meeting with @crouillon asap :)

**Comment by mickaelandrieu** [comment]
@eric-chau for now I apply this patch on ``news-template`` branch in one single commit to remove it when we will have decided how to do.

**Comment by eric-chau** [comment]
@mickaelandrieu ok.

**Comment by mickaelandrieu** [comment]
@crouillon done: ready for merge ;)

File: Controller/FrontController.php [hunk 1]
```diff
@@ -565,9 +565,15 @@ protected function invokeAction($type = self::MASTER_REQUEST)
             $actionArguments = $controllerResolver->getArguments($request, $controller);
 
             $response = call_user_func_array($controller, $actionArguments);
-            $dispatcher->dispatch($eventName.'.postcall', new Event\PostResponseEvent($response, $request));
 
-            return $response;
+            if ($response instanceof Response) {
+                // happens when response have not been already sent
```
Comment (eric-chau) [inline Controller/FrontController.php]: This comment is wrong. ``$response`` still can be instance of ``Response`` and being already sent.

Comment (mickaelandrieu) [inline Controller/FrontController.php]: in which case ?

**Comment by mickaelandrieu** [comment]
closed. I let the pull request on my repository for the record.

**Pull request closed without merging**